router/api_test.go

```diff
@@ -67,11 +106,350 @@ func TestListOutput(t *testing.T) {
+		{"WithFileConfigAndDbPath", "test/cfg.yaml", "", false, "raw", "my-slack", "route1", "database/webhooks.db", ""},
+		{"WithNewConfig", "", "", true, "", "", "", "./webhooks.db", ""},
+		{"WithNewConfigAndDbPath", "test/cfg.yaml", "", true, "", "", "", "./webhooks.db", ""},
+		{"WithPostgresParams", "", "ParamsName", true, "", "", "", "", "postgres://ParamsUser:ParamsPassword@ParamsDbHostName:ParamsPort/ParamsDbName?sslmode=ParamsSslMode"},
+		{"WithPostgresUrl", "", "ParamsName", true, "", "", "", "", "postgres://ParamsUser:ParamsPassword@ParamsDbHostName:ParamsPort/ParamsDbName?sslmode=ParamsSslMode"},
+	}
+	for _, test := range tests {
+		t.Run("test "+test.funcName, func(t *testing.T) {
+			defer Instance().cleanInstance()
+			savedPathToDb := os.Getenv("PATH_TO_DB")
+			savedPostgresUrl := os.Getenv("POSTGRES_URL")
+			os.Setenv("PATH_TO_DB", test.dbPath)
+			os.Setenv("POSTGRES_URL", test.psqlUrl)
+			defer func() {
+				os.Setenv("PATH_TO_DB", savedPathToDb)
+				os.Setenv("POSTGRES_URL", savedPostgresUrl)
+			}()
+
+			err := runFunc(test.funcName, test.cfgPath, test.dbPath, test.tenantName, test.psqlUrl)
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+			if test.clearCfg {
+				assert.Equal(t, 0, len(Instance().templates), "no template expected")
+				assert.Equal(t, 0, len(Instance().outputs), "no output expected")
+				assert.Equal(t, 0, len(Instance().inputRoutes), "no route expected")
+			} else {
+				assert.Equal(t, 1, len(Instance().templates), "one template expected")
+				assert.Contains(t, Instance().templates, test.templateName)
+
+				assert.Equal(t, 1, len(Instance().outputs), "one output expected")
+				assert.Contains(t, Instance().outputs, test.outputName)
+				assert.Equal(t, test.outputName, Instance().outputs[test.outputName].GetName(), "check name failed")
+
+				assert.Equal(t, 1, len(Instance().inputRoutes), "one route expected")
+				assert.Contains(t, Instance().inputRoutes, test.routeName)
+				assert.Contains(t, Instance().inputRoutes[test.routeName].Outputs, test.outputName)
+				assert.Equal(t, test.templateName, Instance().inputRoutes[test.routeName].Template, "one template expected")
+			}
+			if postgresDb, ok := dbservice.Db.(*postgresdb.PostgresDb); ok {
+				assert.Equal(t, test.psqlUrl, postgresDb.ConnectUrl, "url configured")
+				assert.Equal(t, test.tenantName, postgresDb.TenantName, "tenantName configured")
+			}
+
+			if boltDb, ok := dbservice.Db.(*boltdb.BoltDb); ok {
+				assert.Equal(t, test.dbPath, boltDb.DbPath, "dbPath configured")
+			}
+		})
+	}
+}
+
+var cfg = `Name: tenant
+
+routes:
+- name: route1
+  outputs: ["my-slack"]
+  template: raw
+  plugins:
+   Policy-Show-All: true
+
+templates:
+- name: raw
+  body: |
+   package postee
+   result:=input
+
+outputs:
+- name: my-slack
+  type: slack
+  enable: true
+  url: https://hooks.slack.com/services/ABCDF/1234/TTT`
+
+func runFunc(funcName, cfgPath, dbPath, tenantName, psqlUrl string) error {
```

Comment:
no need to swich here. you can assign function to variables and then add function references to test data

Reply:
Done